internal/clients/http/gcp/gcp_client.go

```diff
@@ -133,8 +150,43 @@ func (c *gcpClient) InsertInstances(ctx context.Context, namePattern *string, im
 
 	op, err := client.BulkInsert(ctx, req)
 	if err != nil {
-		return nil, fmt.Errorf("cannot bulk insert instances: %w", err)
+		span.SetStatus(codes.Error, err.Error())
+		log.Trace().Msgf("Bulk insert operation has failed: %v", err)
+		return nil, nil, fmt.Errorf("cannot bulk insert instances: %w", err)
+	}
+	if err := op.Wait(ctx); err != nil {
+		log.Trace().Msgf("Bulk insert operation has failed: %v", err)
+		span.SetStatus(codes.Error, err.Error())
+		return nil, nil, fmt.Errorf("cannot bulk insert instances: %w", err)
 	}
 
-	return ptr.To(op.Name()), nil
+	filter := fmt.Sprintf("labels.rhhcc-id=%v", uuid)
+	lstReq := &computepb.AggregatedListInstancesRequest{
+		Project: c.auth.Payload,
+		Filter:  &filter,
+	}
+
+	if op.Done() {
+		ids := make([]*string, 0)
+		instances := client.AggregatedList(ctx, lstReq)
+		for {
+			pair, err := instances.Next()
+			if errors.Is(err, iterator.Done) {
+				log.Trace().Msg("Instances iterator has finished")
+				break
+			} else if err != nil {
+				log.Trace().Msgf("An error has occured %s", err.Error())
+				span.SetStatus(codes.Error, err.Error())
+				return nil, nil, fmt.Errorf("cannot list instances: %w", err)
+			} else {
+				instances := pair.Value.Instances
+				for _, o := range instances {
+					idAsString := strconv.FormatUint(o.GetId(), 10)
+					ids = append(ids, &idAsString)
+				}
+			}
+		}
+		return ids, ptr.To(op.Name()), nil
+	}
+	return nil, ptr.To(op.Name()), nil
```

Comment:
This case seems to be accessible only when the operation is not done, but given we were waiting for it, I'd consider it a failed state.

This could be IMHO done by

```go
if !op.Done() {
  return nil, ptr.To(op.Name()), <error explaining the failure - the operation name ..... has failed to finish within expected time>
}
```

Reply:
You are right, it is a failure. fixing